### PR TITLE
Test Solidus against Ruby 3.2

### LINKED ISSUE #4814: Support Ruby 3.2

### PR DESCRIPTION
**Pull request by kennyadsl**
Fixes #4814.

## Summary

🎅 brought us a new Ruby version. Is Solidus ready for it?

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [ ] I have written a thorough PR description.
- [ ] I have kept my commits small and atomic.
- [ ] I have used clear, explanatory commit messages.

The following are not always needed (~cross them out~ if they are not):

- [ ] I have added automated tests to cover my changes.
- [ ] I have attached screenshots to demo visual changes.
- [ ] I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- [ ] I have updated the README to account for my changes.

### CODE REVIEW

**Comment by kennyadsl** [comment]
We need to wait for https://github.com/CircleCI-Public/cimg-ruby/pull/105 to have the Ruby 3.2 docker image available.

**Comment by jarednorman** [comment]
A lot of the errors currently look like they relate to keyword handling:

CleanShot 2022-12-31 at 08 59 43

**Comment by jarednorman** [comment]
Possibly relevant:

- https://github.com/rspec/rspec-mocks/pull/1394
- https://github.com/rspec/rspec-mocks/pull/1514

**Comment by waiting-for-dev** [comment]
> A lot of the errors currently look like they relate to keyword handling. Should be easy enough to fix.
> 
> CleanShot 2022-12-31 at 08 59 43

@jarednorman, can you still see those errors? 🤔

**Comment by kennyadsl** [comment]
> can you still see those errors? 🤔

@waiting-for-dev they've been solved when the new version of `rspec-mocks` has been released, see [this comment](https://github.com/solidusio/solidus/pull/4817#issuecomment-1370323342).

**Review by waiting-for-dev — APPROVED** [review]
Thanks, @kennyadsl ❤️

File: .circleci/config.yml [hunk 1]
```diff
@@ -334,7 +334,7 @@ workflows:
       - test_solidus:
           context: slack-secrets
           name: &name "test-rails-<<matrix.rails>>-ruby-<<matrix.ruby>>-<<matrix.database>>-<<#matrix.paperclip>>paperclip<</matrix.paperclip>><<^matrix.paperclip>>activestorage<</matrix.paperclip>>"
-          matrix: { parameters: { rails: ['7.0'], ruby: ['3.0', '3.1'], database: ['mysql', 'sqlite', 'postgres'], paperclip: [true, false] } }
+          matrix: { parameters: { rails: ['7.0'], ruby: ['3.1', '3.2'], database: ['mysql', 'sqlite', 'postgres'], paperclip: [true, false] } }
```
Comment (elia) [inline .circleci/config.yml:337]: Thoughts on keeping 3.0 + rails 7 around?

**Review by elia — APPROVED** [review]
All great! Just one questions, not blocking

**Review by spaghetticode — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (elia) [reply .circleci/config.yml:337]: I think it still would be better, and ensure we have a ruby version tested for adjacent rails versions, e.g. having ruby 3.0 be the shared version between rails 7 and 6.1, but it's just a nice to have.

It's fine if we merge and then add it later 👍👍👍

Comment (kennyadsl) [reply .circleci/config.yml:337]: Ok, that makes a lot of sense. I think we might accomplish the same result by testing Ruby 3.1 compatibility with Rails 6.1.

Comment (kennyadsl) [reply .circleci/config.yml:337]: Done!

**Pull request merged**